extensions/amp-consent/0.1/consent-config.js

```diff
@@ -149,8 +152,59 @@ export class ConsentConfig {
         }
       }
     }
+    return this.mergeGeoOverride_(config).then(mergedConfig =>
+      this.validateMergedGeoOverride_(mergedConfig)
+    );
+  }
 
-    return config;
+  /**
+   * Merge correct geoOverride object into toplevel config.
+   * @param {!JsonObject} config
+   * @return {!Promise<!JsonObject>}
+   */
+  mergeGeoOverride_(config) {
+    const {promise, resolve} = new Deferred();
+
+    if (config['geoOverride']) {
+      Services.geoForDocOrNull(this.element_).then(geoService => {
+        userAssert(
+          geoService,
+          '%s: requires <amp-geo> to use `geoOverride`',
+          TAG
+        );
+        const mergedConfig = map(config);
+
+        const geoGroups = Object.keys(config['geoOverride']);
+        // Stop at the first group that the geoService says we're in and then merge configs.
+        for (let i = 0; i < geoGroups.length; i++) {
+          if (geoService.isInCountryGroup(geoGroups[i]) === GEO_IN_GROUP.IN) {
+            deepMerge(mergedConfig, config['geoOverride'][geoGroups[i]], 1);
+            break;
+          }
+        }
+        delete mergedConfig['geoOverride'];
+        resolve(mergedConfig);
+      });
+    } else {
+      resolve(config);
+    }
+    return promise;
```

Comment:
instead of using `Deferred`, you can do

```js
if (!config['geoOverride']) {
  return Promise.resolve(config)
}

return Services.geoForDocOrNull(this.element_).then(() => {
  ...
  return mergedConfig;
});

```

Reply:
Sweet